app/thirdparty/java/other/flexmark/ext/codeblocks/LineNumbersAttributeProvider.java

```diff
@@ -0,0 +1,33 @@
+package other.flexmark.ext.codeblocks;
+
+import com.vladsch.flexmark.ast.FencedCodeBlock;
+import com.vladsch.flexmark.html.AttributeProvider;
+import com.vladsch.flexmark.html.AttributeProviderFactory;
+import com.vladsch.flexmark.html.IndependentAttributeProviderFactory;
+import com.vladsch.flexmark.html.renderer.AttributablePart;
+import com.vladsch.flexmark.html.renderer.LinkResolverContext;
+import com.vladsch.flexmark.util.ast.Node;
+import com.vladsch.flexmark.util.html.Attributes;
+
+/**
+ * AttributeProvider of flexmark-java extension for prism syntax highlighter.
+ */
+public class LineNumbersAttributeProvider implements AttributeProvider {
```

Comment:
Suggestion: Move LineNumbersAttributeProvider and LineNumbersExtension to one class (with 2 inner classes) together.

Did you write this on your own or is it a copy from some suggestion at some flexmark discussion? If so, attribution is required

Reply:
I referenced codes at https://github.com/riversun/flexmark-ext-prism-linenumber.git. Sorry, I didn't add attribution.

Reply:
@guang-lin  can you also update app/src/main/res/raw/licenses_3rd_party.md ?

Reply:
OK, I will.